pkg/ebpf/bpf_error.go

```diff
@@ -0,0 +1,223 @@
+package ebpf
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+	"unsafe"
+
+	"github.com/aquasecurity/tracee/pkg/logger"
+)
+
+const BPF_MAX_ERR_FILE_LEN = 72
+
+type BPFErrorType uint32
+
+const (
+	BPF_ERR_UNSPEC BPFErrorType = iota
+
+	BPF_ERR_INIT_CONTEXT
+
+	BPF_ERR_MAP_UPDATE_ELEM
+	BPF_ERR_MAP_DELETE_ELEM
+	BPF_ERR_GET_CURRENT_COMM
+	BPF_ERR_TAIL_CALL
+)
+
+var stringMap = map[BPFErrorType]string{
+	BPF_ERR_UNSPEC: "BPF_ERR_UNSPEC",
+
+	BPF_ERR_INIT_CONTEXT: "BPF_ERR_INIT_CONTEXT",
+
+	BPF_ERR_MAP_UPDATE_ELEM:  "BPF_ERR_MAP_UPDATE_ELEM",
+	BPF_ERR_MAP_DELETE_ELEM:  "BPF_ERR_MAP_DELETE_ELEM",
+	BPF_ERR_GET_CURRENT_COMM: "BPF_ERR_GET_CURRENT_COMM",
+	BPF_ERR_TAIL_CALL:        "BPF_ERR_TAIL_CALL",
+}
+
+var errorMap = map[BPFErrorType]string{
+	BPF_ERR_UNSPEC: "Unspecifed BPF error",
+
+	BPF_ERR_INIT_CONTEXT: "Failed to init context",
+
+	BPF_ERR_MAP_UPDATE_ELEM:  "Failed to add or update a map element",
+	BPF_ERR_MAP_DELETE_ELEM:  "Failed to delete a map element",
+	BPF_ERR_GET_CURRENT_COMM: "Failed to get current command",
+	BPF_ERR_TAIL_CALL:        "Failed to tail call",
+}
+
+func (b BPFErrorType) String() string {
+	if bpfErr, found := stringMap[b]; found {
+		return bpfErr
+	}
+
+	return stringMap[BPF_ERR_UNSPEC]
+}
+
+type BPFError struct {
+	ret  int64
+	id   uint32
+	line uint32
+	file [BPF_MAX_ERR_FILE_LEN]byte
```

Comment:
Why file?

Reply:
In the future we'll have bpf code split in more files.